File: ReverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp=[]
        p=head
        while(p):
            temp.append(p.val)
            p=p.next
            
        result=p=ListNode(-1)
        for i in range(len(temp)-1,-1,-1):
            p.next=ListNode(temp[i])
            p=p.next
            
        
        return result.next
        

File: ReverseLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    ListNode curr = null;
    public ListNode reverseList(ListNode head) {
        ListNode p = new ListNode(-1);
        this.curr = p;

        reverseOrder(head);

        return p.next;
    }
    private ListNode reverseOrder(ListNode head){
        if(head == null)
            return null;

        ListNode p = reverseOrder(head.next);
        this.curr.next = new ListNode(head.val);
        this.curr = this.curr.next;

        return head;
    }
}
